src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -82,10 +75,12 @@ internal Tables(Node[] buckets, object[] locks, int[] countPerLock)
         /// </summary>
         private static bool IsValueWriteAtomic()
         {
-            Type valueType = typeof(TValue);
-
             // We don't have Type.IsClass, so we'll check if it's a reference type by seeing if it gets boxed;
             bool isClass = (object)default(TValue) == null;
+            if (isClass)
```

Comment:
And more cleanly than I suggest (too used to `default(TValue?) == null` begin nice when that is what I want to check for).